test/test_utf8.cpp

```diff
@@ -237,8 +292,9 @@ TEST(UTF8_Compare_Core_utf8_zero)
     // Number of trailing 0 makes a difference
     CHECK_EQUAL(true, str_compare(StringData("a\0", 2), StringData("a\0\0", 3)));
     CHECK_EQUAL(false, str_compare(StringData("a\0\0", 3), StringData("a\0", 2)));
-}
 
+    parent.destroy_deep();
```

Comment:
This works, but I'd suggest using a `DeepArrayDestroyGuard` on the parent array at the top instead.